### ISSUE TITLE
Distribution-over-time visualization for gradient/weight evolution

### ISSUE THREAD

**Issue by ryanoneill** [ryanoneill/envision#359]
## Summary

There is no way to visualize how a distribution changes over time — a key visualization for ML training. A single `Histogram` shows one snapshot, but practitioners need to see the evolution of gradient distributions, weight distributions, or activation distributions across training steps.

## Use cases

- **Gradient histograms over training:** TensorBoard's signature visualization. Shows the distribution of gradient values per layer evolving across steps. Reveals vanishing/exploding gradients, training instability, mode collapse.
- **Weight distribution evolution:** How model weights change shape over training. Healthy training shows gradual spreading; pathological training shows concentration or divergence.
- **Activation distribution shifts:** Tracking internal representation changes across epochs (batch norm drift, representation collapse).

These are standard in TensorBoard, Weights & Biases, and MLflow.

## What's needed

Two common visual encodings for this:

### Option A: Heatmap encoding (step × bin → count)
A 2D heatmap where X = training step, Y = value bin, color intensity = count/frequency. This is the most space-efficient and works well in terminals. The existing `Heatmap` component could serve this with the caller doing manual binning, but a purpose-built adapter or convenience constructor would make it practical.

### Option B: Ridgeline / stacked histogram
Multiple histograms stacked vertically (one per time step), slightly offset so they look like a mountain range. More visually intuitive but harder to render in a terminal at useful resolution.

## Suggested approach

Option A (heatmap encoding) is the pragmatic path for a terminal UI:

```rust
// Hypothetical API
let dist_map = DistributionOverTime::new()
    .add_snapshot(step_0, &gradient_values_0)
    .add_snapshot(step_100, &gradient_values_100)
    .add_snapshot(step_200, &gradient_values_200)
    .with_bins(50)
    .with_title("Gradient Distribution: Layer 3");

// Internally computes: for each snapshot, bin the values into a histogram,
// then assemble into a 2D grid (steps × bins) rendered as a Heatmap
```

This could be implemented as:
1. A utility function that converts `Vec<(step, Vec<f64>)>` into `HeatmapState` with auto-binning
2. Or a new lightweight component that wraps `Heatmap` with the binning logic built in

Either way, the heavy lifting (2D color grid rendering) already exists in `Heatmap`.

**Comment by ryanoneill** [comment]
Good idea. This can be composed from the existing Heatmap component today (caller bins the data). A convenience DistributionOverTime adapter is planned for a future release.
